cupy/cuda/thrust.pyx

```diff
@@ -8,7 +8,9 @@ cimport cython  # NOQA
 from libcpp cimport vector
 
 from cupy.cuda cimport common
+from cupy.cuda cimport device
 from cupy.cuda cimport memory
+from cupy.cuda cimport runtime
```

Comment:
```suggestion
from cupy.cuda import runtime
```

Reply:
I don't fully understand why, but switching from `cimport` to `import` here resolved the issue for me.

Reply:
Thanks, Greg! It's indeed confusing...By inspecting the imported C headers (`cupy_common.h` etc) I also thought the problem was this line, so I dug into the C preprocessor output, and oddly all macro guards were set up right but the includes just didn't get expanded: 
```C++
...
# 658 "cupy/cuda/thrust.cpp" 2
# 1 "cupy/cuda/../cuda/cupy_cuda.h" 1
# 54 "cupy/cuda/../cuda/cupy_cuda.h"
# 1 "cupy/cuda/../cuda/cupy_cuda_common.h" 1  // <- not OK: nothing expanded!
# 55 "cupy/cuda/../cuda/cupy_cuda.h" 2  // <- OK, cupy_cuComplex.h is included earlier



# 57 "cupy/cuda/../cuda/cupy_cuda.h"
extern "C" {

bool hip_environment = false;






CUresult cuGetErrorName(...) {
    return CUDA_SUCCESS;
}
...
```

I also found that the order of `cimport common` and `cimport runtime` matters. By switching them I got different errors. Usually this means there's some circular import at the C level, but I don't immediately see where the circularity is...

Reply:
The guard name `INCLUDE_GUARD_CUPY_CUDA_COMMON_H` is reused in two headers 🤬

Reply:
e673795 fixed it.

Reply:
I am amazed it managed to compile until now, lol